ocaml/xxhash/lib/XXHash.ml

```diff
@@ -0,0 +1,61 @@
+exception XXH_error
+
+module type XXHASH = sig
+  type hash
+  type state
+
+  val hash : ?seed:hash -> string -> hash
+  val create : unit -> state
+  val reset : ?seed:hash -> state -> unit
+  val update : state -> string -> unit
+  val digest : state -> hash
+  val free : state -> unit
+
+  val with_state : ?seed:hash -> (state -> unit) -> hash
+end
+
+let check errorcode =
+  if errorcode != 0 then raise XXH_error
+
+module XXHash (Bindings : Xxhash_bindings.BINDINGS) = struct
```

Comment:
This is module `XXHash.XXHash`. It would be more idiomatic to call this `Make`, such that we get `XXHash.Make(Bindings)`.